pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -274,13 +274,20 @@ allroutes:
 			}
 			break allroutes // we have a rule with catch all match prefix: /. Other rules are of no use
 		} else {
-			for _, match := range http.Match {
+			catchall := false
+			if match := catchAllMatch(http); match != nil {
+				// We have a catch all route - see if it is valid and if yes, do not build other routes.
 				if r := translateRoute(push, node, http, match, listenPort, virtualService, serviceRegistry, gatewayNames); r != nil {
+					catchall = true
 					out = append(out, r)
-					rType, _ := getEnvoyRouteTypeAndVal(r)
```

Comment:
where is the break statement to break out of allroutes? once you hit a catch all match, there is no point in processing other routes right?

Reply:
That is correct. First we pick the catchall match if it available and then only build the route for it and use it, if it is not, it just loops through all matches.

Reply:
Let me clarify.

```yaml
vs:
http:
- match
   - uri: /scooby
   route: scooby
- match
   - uri: /
   route: default
- match
   - uri: /foo
   - uri: /bar
   - uri: / # this is what you are handling
   - uri: /bazoo # will be skipped
   route: go to blah
```

The code as it stands before your PR, will see the second match doing a /, and will not even process the third match. With your PR, will process the second and then move onto the third match wherein it processes foo, bar and /, and omits bazoo.

See the difference? You need a `if catchAll, break allroutes`

Reply:
Sorry,. Was thinking some thing else, yes, it needs to break, Fixed it.